oximeter/impl/src/schema/mod.rs

```diff
@@ -153,18 +159,58 @@ fn validate_timeseries_name(s: &str) -> Result<&str, MetricsError> {
     }
 }
 
+/// Text descriptions for the target and metric of a timeseries.
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+pub struct TimeseriesDescription {
+    pub target: String,
+    pub metric: String,
```

Comment:
These fields being `pub` means a client can create `TimeseriesDescription`s that aren't valid (e.g., a target containing `:`). Is that okay? (Same question for the derived `Deserialize` impl, which will allow any strings)

Reply:
The _description_ is just freeform text. The _name_ cannot have `:` in it, but that's checked in other ways. I don't have any reason to restrict the format of the description at this time, except possibly limiting the total length?

Reply:
Oh, yeah, I completely misunderstood this (which you realized in the subsequent comment, haha). Sorry! Ignore this suggestion.